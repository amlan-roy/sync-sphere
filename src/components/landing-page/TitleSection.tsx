import React from "react";

type TitleSectionProps = {
  title: string;
  subheading?: string;
  pill: string;
};

const TitleSection: React.FC<TitleSectionProps> = ({
  title,
  subheading,
  pill,
}) => {
  return (
    <React.Fragment>
      <section
        className="
    flex flex-col gap-4 justify-center items-start
    md:items-center
    "
      >
        <article
          data-testid="title-section__pill"
          className="
            rounded-full p-[1px] text-sm
            dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple
            "
        >
          <div
            className="
            rounded-full px-3 py-1
            dark:bg-black
            "
          >
            {pill}
          </div>
        </article>
        {subheading ? (
          <>
            <h2
              data-testid="title-section__title"
              className="text-left text-3xl font-semibold
            md:text-centre
            sm:text-5xl"
            >
              {title}
            </h2>
            <p
              data-testid="title-section__subheading"
              className="dark:text-washed-purple-700
        sm:max-w-[450px]
        md:text-center"
            >
              {subheading}
            </p>
          </>
        ) : (
          <h1
            data-testid="title-section__title"
            className="text-left text-4xl font-semibold
            md:text-centre
            sm:text-6xl"
          >
            {title}
          </h1>
        )}
      </section>
    </React.Fragment>
  );
};
export default TitleSection;
