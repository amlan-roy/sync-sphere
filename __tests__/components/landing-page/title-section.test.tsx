/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import TitleSection from "@/components/landing-page/TitleSection";

describe("components/landing-page/TitleSection", () => {
  const defaultProps = {
    title: "Default Title Text",
    pill: "Default Pill Text",
  };
  it("renders a title and pill text", () => {
    const tree = render(<TitleSection {...defaultProps} />);

    expect(tree.getByTestId("title-section__title")).toHaveTextContent(
      defaultProps.title
    );
    expect(tree.getByTestId("title-section__pill")).toHaveTextContent(
      defaultProps.pill
    );
    expect(tree).toMatchSnapshot();
  });
  it("renders a title, subheading, pill text", () => {
    const tree = render(
      <TitleSection {...defaultProps} subheading="Default Subheading Text" />
    );

    expect(tree.getByTestId("title-section__title")).toHaveTextContent(
      defaultProps.title
    );
    expect(tree.getByTestId("title-section__subheading")).toHaveTextContent(
      "Default Subheading Text"
    );
    expect(tree.getByTestId("title-section__pill")).toHaveTextContent(
      defaultProps.pill
    );
    expect(tree).toMatchSnapshot();
  });
});
