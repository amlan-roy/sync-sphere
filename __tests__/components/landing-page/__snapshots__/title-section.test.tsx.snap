// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a title and pill text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="   flex flex-col gap-4 justify-center items-start   md:items-center   "
      >
        <article
          class="   rounded-full p-[1px] text-sm   dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple   "
          data-testid="title-section__pill"
        >
          <div
            class="   rounded-full px-3 py-1   dark:bg-black   "
          >
            Default Pill Text
          </div>
        </article>
        <h1
          class="text-left text-4xl font-semibold   md:text-centre   sm:text-6xl"
          data-testid="title-section__title"
        >
          Default Title Text
        </h1>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="   flex flex-col gap-4 justify-center items-start   md:items-center   "
    >
      <article
        class="   rounded-full p-[1px] text-sm   dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple   "
        data-testid="title-section__pill"
      >
        <div
          class="   rounded-full px-3 py-1   dark:bg-black   "
        >
          Default Pill Text
        </div>
      </article>
      <h1
        class="text-left text-4xl font-semibold   md:text-centre   sm:text-6xl"
        data-testid="title-section__title"
      >
        Default Title Text
      </h1>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home renders a title, subheading, pill text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="   flex flex-col gap-4 justify-center items-start   md:items-center   "
      >
        <article
          class="   rounded-full p-[1px] text-sm   dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple   "
          data-testid="title-section__pill"
        >
          <div
            class="   rounded-full px-3 py-1   dark:bg-black   "
          >
            Default Pill Text
          </div>
        </article>
        <h2
          class="text-left text-3xl font-semibold   md:text-centre   sm:text-5xl"
          data-testid="title-section__title"
        >
          Default Title Text
        </h2>
        <p
          class="dark:text-washed-purple-700   sm:max-w-[450px]   md:text-center"
          data-testid="title-section__subheading"
        >
          Default Subheading Text
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="   flex flex-col gap-4 justify-center items-start   md:items-center   "
    >
      <article
        class="   rounded-full p-[1px] text-sm   dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple   "
        data-testid="title-section__pill"
      >
        <div
          class="   rounded-full px-3 py-1   dark:bg-black   "
        >
          Default Pill Text
        </div>
      </article>
      <h2
        class="text-left text-3xl font-semibold   md:text-centre   sm:text-5xl"
        data-testid="title-section__title"
      >
        Default Title Text
      </h2>
      <p
        class="dark:text-washed-purple-700   sm:max-w-[450px]   md:text-center"
        data-testid="title-section__subheading"
      >
        Default Subheading Text
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/landing-page/TitleSection renders a title and pill text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="   flex flex-col gap-4 justify-center items-start   md:items-center   "
      >
        <article
          class="   rounded-full p-[1px] text-sm   dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple   "
          data-testid="title-section__pill"
        >
          <div
            class="   rounded-full px-3 py-1   dark:bg-black   "
          >
            Default Pill Text
          </div>
        </article>
        <h1
          class="text-left text-4xl font-semibold   md:text-centre   sm:text-6xl"
          data-testid="title-section__title"
        >
          Default Title Text
        </h1>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="   flex flex-col gap-4 justify-center items-start   md:items-center   "
    >
      <article
        class="   rounded-full p-[1px] text-sm   dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple   "
        data-testid="title-section__pill"
      >
        <div
          class="   rounded-full px-3 py-1   dark:bg-black   "
        >
          Default Pill Text
        </div>
      </article>
      <h1
        class="text-left text-4xl font-semibold   md:text-centre   sm:text-6xl"
        data-testid="title-section__title"
      >
        Default Title Text
      </h1>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/landing-page/TitleSection renders a title, subheading, pill text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="   flex flex-col gap-4 justify-center items-start   md:items-center   "
      >
        <article
          class="   rounded-full p-[1px] text-sm   dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple   "
          data-testid="title-section__pill"
        >
          <div
            class="   rounded-full px-3 py-1   dark:bg-black   "
          >
            Default Pill Text
          </div>
        </article>
        <h2
          class="text-left text-3xl font-semibold   md:text-centre   sm:text-5xl"
          data-testid="title-section__title"
        >
          Default Title Text
        </h2>
        <p
          class="dark:text-washed-purple-700   sm:max-w-[450px]   md:text-center"
          data-testid="title-section__subheading"
        >
          Default Subheading Text
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="   flex flex-col gap-4 justify-center items-start   md:items-center   "
    >
      <article
        class="   rounded-full p-[1px] text-sm   dark:bg-gradient-to-r dark:from-brand-primaryBlue dark:to-brand-primaryPurple   "
        data-testid="title-section__pill"
      >
        <div
          class="   rounded-full px-3 py-1   dark:bg-black   "
        >
          Default Pill Text
        </div>
      </article>
      <h2
        class="text-left text-3xl font-semibold   md:text-centre   sm:text-5xl"
        data-testid="title-section__title"
      >
        Default Title Text
      </h2>
      <p
        class="dark:text-washed-purple-700   sm:max-w-[450px]   md:text-center"
        data-testid="title-section__subheading"
      >
        Default Subheading Text
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
